### PR TITLE
GH-629 - Property Popover in Kanban view

### PR DESCRIPTION
**Pull request by ditsemto**
#### Summary

Adds a tool-tip popover above the elements of a Kanban card.
The property name is shown in a tool-tip above the element.
Only readonly elements show the tooltip.

The style was added to the main project stylesheet.
A data attribute is used as the tooltip text.

#### Ticket Link

Fixes https://github.com/mattermost/focalboard/issues/629

### CODE REVIEW

**Comment by hahmadia** [comment]
Hi @ditsemto ,

Could you please fix the broken tests that are failing in the CI pipeline? Thanks!

**Comment by ditsemto** [comment]
Hi @hahmadia, the power has been out in my area for the last 6 hours. I apologize for the inconvenience.

I'll get on at as soon as it is back.

File: webapp/src/components/properties/link/link.tsx
Comment (harshilsharma63) [inline webapp/src/components/properties/link/link.tsx:23]: This code would repeat over and over in any new property component that gets added in the future. Also since the logic for adding and generation tooltip is independent of property type, I'd like to remove it from each individual component.

Instead, I think we can add a wrapper `div` with tooltip over here - 

https://github.com/mattermost/focalboard/blob/6cd5003a7aa67ff11fb1bcaca4a46914a8730428/webapp/src/components/kanban/kanbanCard.tsx#L78

This is where the visible properties are added to the kanban card component. That part of the code is only used for readonly view so we can skip the logic of deciding whether to add the tooltip or not. This will have the following advantages -

1. No repeating code in each property component (there are a bunch of new ones being added in the open PRs)
2. Tooltips for new types gets added automatically without any code change.

This approach can also be applied to the gallery view which suffers from the same problem.

File: webapp/src/styles/main.scss
Comment (harshilsharma63) [inline webapp/src/styles/main.scss:244]: Great work here!

Can we move this to a separate file to keeps things better structured and managable?

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]
Awesome work @ditsemto 🎉 I really love how the cards look with the tooltip.

I've added a change request for the approach we've taken here for adding the tooltips. I think with the newly suggested approach the code changes will be less than half what we have right now.

Let me know if you have any questions.

File: webapp/src/components/properties/link/link.tsx
Comment (ditsemto) [reply webapp/src/components/properties/link/link.tsx:23]: Thanks @harshilsharma63, this is a better way of handling it, I'll will update it.

File: webapp/src/styles/main.scss
Comment (ditsemto) [reply webapp/src/styles/main.scss:244]: Will do :+1:

File: webapp/src/components/properties/link/link.tsx
Comment (ditsemto) [reply webapp/src/components/properties/link/link.tsx]: Hi @harshilsharma63,

With regards to adding the same functionality to the gallery view, should one create a new ticket for it so there is a track record of the change or should I add it while I am busy?

Currently I only updated the current Kanban card.

File: webapp/src/widgets/tooltip.tsx
Comment (hahmadia) [inline webapp/src/widgets/tooltip.tsx:23]: Could we move `{props.children}` to the next line?

File: webapp/src/widgets/tooltip.tsx
Comment (hahmadia) [inline webapp/src/widgets/tooltip.tsx:8]: Let's rename this to `title`
```suggestion
    title: string
```

File: webapp/src/components/kanban/kanbanCard.tsx
Comment (hahmadia) [inline webapp/src/components/kanban/kanbanCard.tsx:81]:
```suggestion
                    title={template.name}
```

File: webapp/src/widgets/tooltip.tsx
Comment (hahmadia) [inline webapp/src/widgets/tooltip.tsx:9]: I believe this shouldn't be `any` but instead `React.ReactNode`.

File: webapp/src/widgets/tooltip.tsx
Comment (hahmadia) [inline webapp/src/widgets/tooltip.tsx:17]: Can be simplified
```suggestion
    const placement =  props.placement || 'top'
```

**Review by hahmadia — CHANGES_REQUESTED** [review]
I tried it and it's great! I love it! Thanks a lot!
I just have some minor nit changes to make but usability is awesome. Thanks!

File: webapp/src/components/properties/link/link.tsx
Comment (hahmadia) [reply webapp/src/components/properties/link/link.tsx]: If it's not too much extra work, I think getting gallery view done within this same PR would be great unless you are busy and don't have the time to do it which we can create a new PR for it. Thoughts @harshilsharma63? I'll make it your call but I ultimately think it makes sense to do it one PR.

Comment (harshilsharma63) [reply webapp/src/components/properties/link/link.tsx]: @ditsemto sorry I missed your reply. If it's a quick thing then we can add it for gallery view as well. If not, we can handle it separately.

File: webapp/src/widgets/tooltip.tsx
Comment (ditsemto) [reply webapp/src/widgets/tooltip.tsx:23]: Will do :+1:

File: webapp/src/widgets/tooltip.tsx
Comment (ditsemto) [reply webapp/src/widgets/tooltip.tsx:8]: Sure thing

File: webapp/src/widgets/tooltip.tsx
Comment (ditsemto) [reply webapp/src/widgets/tooltip.tsx:9]: Thanks for this

File: webapp/src/widgets/tooltip.tsx
Comment (ditsemto) [reply webapp/src/widgets/tooltip.tsx:17]: Definitely cleaner thanks :smile:

File: webapp/src/components/properties/link/link.tsx
Comment (ditsemto) [reply webapp/src/components/properties/link/link.tsx]: @harshilsharma63 No worries. 

I made the requested changes on one commit and updated the gallery view in the second commit.

**Review by harshilsharma63 — APPROVED** [review]
Awesome work!

LGTM 🎉

**Comment by harshilsharma63** [comment]
@hahmadia the changes look good to me. We can merge once you approve.

**Review by hahmadia — APPROVED** [review]

**Comment by hahmadia** [comment]
Thanks!!

**Pull request merged**
